uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Import/Parser/Base.php

```diff
@@ -0,0 +1,351 @@
+      $field['dataPattern'] = CRM_Utils_Array::value('dataPattern', $field, '//');
+      $field['headerPattern'] = CRM_Utils_Array::value('headerPattern', $field, '//');
+      $this->addField($name, $field['title'], $field['type'], $field['headerPattern'], $field['dataPattern']);
+    }
+
+    $this->setActiveFields($this->_mapperKeys);
+    $this->absenceTypes = $this->getAbsenceTypes();
+    $this->absenceStatuses = array_flip(LeaveRequest::buildOptions('status_id'));
+  }
+
+  /**
+   * Handle the values in mapField mode.
+   * This function does practically nothing because we don't need to do any validation
+   * during the mapping field phase. However the base parser class calls this method
+   * during MAPFIELD mode and this class needs to implement the method.
+   *
+   * @param array $values
+   *   The array of values belonging to this line.
+   *
+   * @return bool
+   */
+  public function mapField(&$values) {
+    return CRM_Import_Parser::VALID;
+  }
+
+  /**
+   * Handle the values in preview mode.
+   *
+   * @param array $values
+   *   The array of values belonging to this line.
+   *
+   * @return bool
+   *   the result of this processing
+   */
+  public function preview(&$values) {
+    return $this->summary($values);
+  }
+
+  /**
+   * Handle the values in summary mode.
+   *
+   * @param array $values
+   *   The array of values belonging to this line.
+   *
+   * @return bool
+   *   the result of this processing
+   */
+  public function summary(&$values) {
+    $erroneousField = NULL;
+    $this->setActiveFieldValues($values, $erroneousField);
+    $params = &$this->getActiveFieldParams();
+    $errorMessage = NULL;
+    $errorMessage .= $this->validateFieldTypes($params);
+    $errorMessage .= $this->validateOptionValues($params);
+
+    if (!empty($errorMessage)) {
+      array_unshift($values, $errorMessage);
+      $errorMessage = NULL;
+      return CRM_Import_Parser::ERROR;
+    }
+
+    return CRM_Import_Parser::VALID;
+  }
+
+  /**
+   * Handle the values in import mode.
+   * Each row represents a Leave request date with an absence_id property
+   * that identifies leave dates belonging to the same parent leave request.
+   * Information such as the start date and end date of the leave request is replicated across all the individual
+   * leave request dates for the leave request.
+   * The Leave request is created from the first individual leave request date encountered;
+   * Comments, Leave Dates for all the leave request dates and balance change
+   * for that particular leave request date is also created.
+   * Balance changes only will be created for subsequent rows of leave request dates for the already created leave request.
+   *
+   * If there is an error while creating the leave request from the first leave request date encountered,
+   * All other leave request dates will not be considered and the error reported is reported for all the leave dates.
```

Comment:
Thanks for adding the description. I see that each line has a different lenght. Can you please make sure each that each one have +- the same length?